File: csharp/DesiArchitect.DesignPatterns/04_Decorator/After/PackagingDecorator.cs
namespace DesiArchitect.DesignPatterns.Decorator.After;

public class PackagingDecorator : IBill
{
    private readonly IBill _inner;
    private readonly decimal _charge = 15m;

    public PackagingDecorator(IBill inner) => _inner = inner;

    public decimal Total()
    {
        var subtotal = _inner.Total();
        var result = subtotal + _charge;
        Console.WriteLine($"  📦 + Packaging: ₹{_charge} -> ₹{result}");
        return result;
    }
}

File: typescript/src/04-decorator/after/packaging-decorator.ts
import type { IBill } from './ibill.js';

export class PackagingDecorator implements IBill {
  private readonly charge = 15;

  constructor(private readonly inner: IBill) {}

  total(): number {
    const subtotal = this.inner.total();
    const result = subtotal + this.charge;
    console.log(`  📦 + Packaging: ₹${this.charge} -> ₹${result}`);
    return result;
  }
}